### PR TITLE
Add migration of Metric learning for image similarity search using TensorFlow Similarity tutorial with keras 3.

### PR DESCRIPTION
**Pull request by maitry63**
This PR migrates the metric learning implementation for image similarity search from TensorFlow to Keras 3, making it backend-agnostic by replacing all TensorFlow and TF Similarity dependencies with pure Keras APIs.

Tutorial link - https://keras.io/examples/vision/metric_learning_tf_similarity/

Colab - [Notebook](https://colab.sandbox.google.com/gist/maitry63/6ef83733aa36f96633a73632944ec45b/metric-learning-for-image-similarity-search.ipynb#scrollTo=DwT0pPZ-ZI2V)

### CODE REVIEW

**Comment by maitry63** [comment]
@gemini-code-assist review

**Comment by maitry63** [comment]
@gemini-code-assist review

**Comment by maitry63** [comment]
@gemini-code-assist review
